src/node_contextify.cc

```diff
@@ -951,6 +952,125 @@ class ContextifyScript : public BaseObject {
 };
 
 
+void ContextifyContext::CompileFunction(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+
+  // Argument 1: source code
+  CHECK(args[0]->IsString());
+  Local<String> code = args[0].As<String>();
+
+  // Argument 2: filename
+  CHECK(args[1]->IsString());
+  Local<String> filename = args[1].As<String>();
+
+  // Argument 3: line offset
+  CHECK(args[2]->IsNumber());
+  Local<Integer> line_offset = args[2].As<Integer>();
```

Comment:
The Number type is indeed checked in JavaScript, but not that the number is an integer. That should be added.

Reply:
Agreed, will throw in `Number.isInteger` checks.

Reply:
BTW you should use [`validateInteger()`](https://github.com/nodejs/node/blob/7951e6d26b2c2036a8d6dbb92b395f9e15d90e14/lib/internal/validators.js#L62-L76) in `internal/validators.js`.